slurm_ops_manager/slurm_deb_manager.py

```diff
@@ -47,24 +47,22 @@ def _install_slurm_from_deb(self) -> bool:
 
         Returns True on success and False otherwise.
         """
-
         slurm_component = self._slurm_component
-
-        with open("/etc/apt/sources.list.d/bullseye.list", "w") as afile:
-            afile.write("deb http://deb.debian.org/debian bullseye main")
-
         subprocess.call(["apt-get", "install", "--yes", "debian-keyring"])
 
-        # The following keys are carried with the charm to remedy scenarios
+        # The following keys are needed to install pkgs from debian bullseye.
+        # These keys are carried with the charm to remedy scenarios
         # where keyserver.ubuntu.com is not reachable.
         # Ref: https://github.com/omnivector-solutions/slurm-charms/issues/111
-        template_dir = Path(
-            os.path.dirname(os.path.abspath(__file__))) / 'templates'
-        key_one = template_dir / "04EE7237B7D453EC.key"
-        key_two = template_dir / "648ACFD622F3D138.key"
-
-        subprocess.call(["apt-key", "add", key_one.as_posix()])
-        subprocess.call(["apt-key", "add", key_two.as_posix()])
+        # Iterate over the two keys, adding them to the system.
+        for keyfile in ["04EE7237B7D453EC.key", "648ACFD622F3D138.key"]:
+            key = TEMPLATE_DIR / keyfile
+            subprocess.call(["apt-key", "add", key.as_posix()])
+
+        # Add the bullseye repo and run apt update.
+        Path("/etc/apt/sources.list.d/bullseye.list").write_text(
+            "deb http://deb.debian.org/debian bullseye main"
+        )
```

Comment:
this should be in the previous place? Before installing the keyring?

Reply:
https://wiki.debian.org/DebianKeyring - keyring is only used for storing gpg keys, we just need it installed before we add the keys.